File: static/js/admin/admin_courses_approval.js
document.addEventListener('DOMContentLoaded', function() {
    // Logout Modal Functionality
    $('#logout-trigger').click(function(e) {
        e.preventDefault();
        $('#logout-modal').fadeIn();
    });
    
    $('.close-modal').click(function() {
        $('#logout-modal').fadeOut();
        closeCourseModal();
    });
    
    $(window).click(function(e) {
        if ($(e.target).is('#logout-modal')) {
            $('#logout-modal').fadeOut();
        }
        if ($(e.target).is('#courseModal')) {
            closeCourseModal();
        }
    });
    
    $('#confirm-logout').click(function() {
        window.location.href = "/logout";
    });

    // View button functionality
    document.querySelectorAll('.view-btn').forEach(button => {
        button.addEventListener('click', function() {
            const courseId = this.getAttribute('data-course-id');
            showCourseDetails(courseId);
        });
    });

    // Close modal with overlay click
    document.querySelector('.modal-overlay').addEventListener('click', closeCourseModal);

    // Approve button in modal
    document.getElementById('approveBtn').addEventListener('click', function() {
        const courseId = this.getAttribute('data-course-id');
        approveCourse(courseId);
    });

    // Reject button in modal
    document.getElementById('rejectBtn').addEventListener('click', function() {
        const courseId = this.getAttribute('data-course-id');
        rejectCourse(courseId);
    });

    // Search functionality
    const searchInput = document.querySelector('.search-input');
    if (searchInput) {
        searchInput.addEventListener('input', (e) => {
            const searchTerm = e.target.value.toLowerCase();
            const rows = document.querySelectorAll('tbody tr');
            
            rows.forEach(row => {
                const text = row.textContent.toLowerCase();
                row.style.display = text.includes(searchTerm) ? '' : 'none';
            });
        });
    }

    // Filter functionality
    const filterSelect = document.querySelector('.filter-select');
    if (filterSelect) {
        filterSelect.addEventListener('change', (e) => {
            const filterValue = e.target.value;
            const rows = document.querySelectorAll('tbody tr');
            
            rows.forEach(row => {
                const categoryElement = row.querySelector('.course-category');
                if (categoryElement) {
                    const category = categoryElement.textContent.toLowerCase();
                    if (filterValue === 'all' || category.includes(filterValue)) {
                        row.style.display = '';
                    } else {
                        row.style.display = 'none';
                    }
                }
            });
        });
    }
});

function showCourseDetails(courseId) {
    fetch(`/course/details/${courseId}`)
        .then(response => response.json())
        .then(data => {
            if(data.status === 'success') {
                const c = data.course;
                document.getElementById('courseDetails').innerHTML = `
                    <div class="detail-section">
                        <h4>Course Information</h4>
                        <div class="detail-grid">
                            <div class="detail-item">
                                <span class="detail-label">Course Code</span>
                                <span class="detail-value">${c.course_code}</span>
                            </div>
                            <div class="detail-item">
                                <span class="detail-label">Title</span>
                                <span class="detail-value">${c.course_title}</span>
                            </div>
                            <div class="detail-item">
                                <span class="detail-label">Category</span>
                                <span class="detail-value">${c.course_category}</span>
                            </div>
                            <div class="detail-item">
                                <span class="detail-label">Status</span>
                                <span class="detail-value status-badge status-pending">${c.course_status}</span>
                            </div>
                            <div class="detail-item">
                                <span class="detail-label">Duration</span>
                                <span class="detail-value">${c.duration_hours} hours</span>
                            </div>
                            <div class="detail-item">
                                <span class="detail-label">Fee</span>
                                <span class="detail-value">₱${parseFloat(c.course_fee).toFixed(2)}</span>
                            </div>
                            <div class="detail-item">
                                <span class="detail-label">Max Students</span>
                                <span class="detail-value">${c.max_students}</span>
                            </div>
                        </div>
                    </div>
                    
                    <div class="detail-section">
                        <h4>Description</h4>
                        <div class="detail-content">${c.course_description || 'No description provided'}</div>
                    </div>
                    
                    <div class="detail-section">
                        <h4>Target Audience</h4>
                        <div class="detail-content">${c.target_audience || 'No target audience specified'}</div>
                    </div>
                    
                    <div class="detail-section">
                        <h4>Prerequisites</h4>
                        <div class="detail-content">${c.prerequisites || 'No prerequisites required'}</div>
                    </div>
                    
                    <div class="detail-section">
                        <h4>Learning Outcomes</h4>
                        <div class="detail-content">${c.learning_outcomes || 'No learning outcomes specified'}</div>
                    </div>
                    
                    <div class="detail-section">
                        <h4>Submitted By</h4>
                        <div class="submitted-by">
                            <div class="avatar">${c.first_name[0]}${c.last_name[0]}</div>
                            <div>
                                <div class="instructor-name">${c.first_name} ${c.last_name}</div>
                                <div class="instructor-details">
                                    <span>${c.role}</span>
                                    <span>•</span>
                                    <span>ID: ${c.user_id}</span>
                                </div>
                            </div>
                        </div>
                    </div>
                `;
                document.getElementById('approveBtn').setAttribute('data-course-id', c.course_id);
                document.getElementById('rejectBtn').setAttribute('data-course-id', c.course_id);
                openCourseModal();
            } else {
                Swal.fire({
                    title: 'Error',
                    text: data.message || 'Failed to load course details',
                    icon: 'error',
                    confirmButtonColor: '#0033A0'
                });
            }
        })
        .catch(error => {
            console.error('Error fetching course details:', error);
            Swal.fire({
                title: 'Error',
                text: 'Failed to load course details. Please try again.',
                icon: 'error',
                confirmButtonColor: '#0033A0'
            });
        });
}

function openCourseModal() {
    document.getElementById('courseModal').classList.add('active');
    document.body.style.overflow = 'hidden';
}

function closeCourseModal() {
    document.getElementById('courseModal').classList.remove('active');
    document.body.style.overflow = 'auto';
}

async function approveCourse(courseId) {
    // Close the course modal first
    closeCourseModal();
    
    // Small delay to ensure modal is closed before showing SweetAlert
    await new Promise(resolve => setTimeout(resolve, 100));
    
    const result = await Swal.fire({
        title: 'Approve this course?',
        text: 'This will make the course available to all users',
        icon: 'question',
        showCancelButton: true,
        confirmButtonColor: '#0033A0',
        cancelButtonColor: '#6b7280',
        confirmButtonText: 'Yes, approve',
        cancelButtonText: 'Cancel',
        allowOutsideClick: false,
        // Remove reverseButtons to keep Cancel on the right
        reverseButtons: false
    });

    if (result.isConfirmed) {
        try {
            const response = await fetch(`/course/approve/${courseId}`, { 
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                }
            });
            const data = await response.json();
            if (data.status === 'success') {
                await Swal.fire({
                    title: 'Approved!',
                    text: data.message,
                    icon: 'success',
                    confirmButtonColor: '#0033A0'
                });
                // Remove the course row from table
                const courseRow = document.getElementById(`course-${courseId}`);
                if (courseRow) {
                    courseRow.remove();
                }
                
                // Check if table is now empty
                const remainingRows = document.querySelectorAll('tbody tr');
                if (remainingRows.length === 0) {
                    setTimeout(() => {
                        location.reload();
                    }, 1000);
                }
            } else {
                await Swal.fire({
                    title: 'Error',
                    text: data.message || 'Failed to approve course',
                    icon: 'error',
                    confirmButtonColor: '#0033A0'
                });
                // Reopen the modal if there was an error
                showCourseDetails(courseId);
            }
        } catch (error) {
            console.error('Error approving course:', error);
            await Swal.fire({
                title: 'Error',
                text: 'Request failed: ' + error.message,
                icon: 'error',
                confirmButtonColor: '#0033A0'
            });
            // Reopen the modal if there was an error
            showCourseDetails(courseId);
        }
    } else {
        // If user cancels, reopen the modal
        showCourseDetails(courseId);
    }
}

async function rejectCourse(courseId) {
    // Close the course modal first
    closeCourseModal();
    
    // Small delay to ensure modal is closed before showing SweetAlert
    await new Promise(resolve => setTimeout(resolve, 100));
    
    const result = await Swal.fire({
        title: 'Reject this course?',
        text: 'This will permanently remove the course from the system',
        icon: 'warning',
        showCancelButton: true,
        confirmButtonColor: '#d33',
        cancelButtonColor: '#6b7280',
        confirmButtonText: 'Yes, reject',
        cancelButtonText: 'Cancel',
        allowOutsideClick: false,
        // Remove reverseButtons to keep Cancel on the right
        reverseButtons: false
    });

    if (result.isConfirmed) {
        try {
            const response = await fetch(`/course/reject/${courseId}`, { 
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                }
            });
            const data = await response.json();
            if (data.status === 'success') {
                await Swal.fire({
                    title: 'Rejected!',
                    text: data.message,
                    icon: 'success',
                    confirmButtonColor: '#0033A0'
                });
                // Remove the course row from table
                const courseRow = document.getElementById(`course-${courseId}`);
                if (courseRow) {
                    courseRow.remove();
                }
                
                // Check if table is now empty
                const remainingRows = document.querySelectorAll('tbody tr');
                if (remainingRows.length === 0) {
                    setTimeout(() => {
                        location.reload();
                    }, 1000);
                }
            } else {
                await Swal.fire({
                    title: 'Error',
                    text: data.message || 'Failed to reject course',
                    icon: 'error',
                    confirmButtonColor: '#0033A0'
                });
                // Reopen the modal if there was an error
                showCourseDetails(courseId);
            }
        } catch (error) {
            console.error('Error rejecting course:', error);
            await Swal.fire({
                title: 'Error',
                text: 'Request failed: ' + error.message,
                icon: 'error',
                confirmButtonColor: '#0033A0'
            });
            // Reopen the modal if there was an error
            showCourseDetails(courseId);
        }
    } else {
        // If user cancels, reopen the modal
        showCourseDetails(courseId);
    }
}

// Escape key to close modals
document.addEventListener('keydown', function(e) {
    if (e.key === 'Escape') {
        closeCourseModal();
        $('#logout-modal').fadeOut();
    }
});

File: admin_courses_approval.py
from flask import Blueprint, render_template, request, jsonify, session, flash, redirect, url_for
from database import get_db
from datetime import datetime

admin_courses_approval_bp = Blueprint('admin_courses_approval', __name__)

@admin_courses_approval_bp.route('/courses/pending', methods=['GET'])
def view_pending_courses():
    if 'user_id' not in session or session.get('role') != 'admin':
        flash('You need to login as admin first', 'error')
        return redirect(url_for('login.login_page'))

    db = get_db()
    cursor = db.cursor(dictionary=True)

    try:
        # --- Fetch pending courses ---
        query = """
            SELECT 
                c.course_id, c.course_code, c.course_title, c.course_status,
                l.user_id, l.role,
                p.first_name, p.last_name,
                c.course_description, c.course_category, c.target_audience,
                c.prerequisites, c.learning_outcomes, c.duration_hours,
                c.course_fee, c.max_students, c.published, c.prerequisites,
                c.created_by, c.approved_by, c.date_created, c.learning_outcomes,
                c.date_updated, c.date_published, c.date_modified
            FROM courses c
            JOIN login l ON c.created_by = l.user_id
            JOIN personal_information p ON l.user_id = p.user_id
            WHERE c.course_status = 'pending' AND l.role = 'staff'
        """
        cursor.execute(query)
        courses = cursor.fetchall()

        # --- Fetch admin profile picture ---
        profile_picture = 'default.png'
        try:
            cursor.execute("""
                SELECT profile_picture 
                FROM personal_information 
                WHERE user_id = %s
            """, (session['user_id'],))
            result = cursor.fetchone()
            if result and result.get('profile_picture'):
                profile_picture = result['profile_picture']
        except Exception:
            profile_picture = 'default.png'

        return render_template(
            'admin/admin_courses_approval.html',
            courses=courses,
            profile_picture=profile_picture
        )

    except Exception as e:
        return str(e), 500
    finally:
        cursor.close()


@admin_courses_approval_bp.route('/course/details/<int:course_id>', methods=['GET'])
def get_course_details(course_id):
    try:
        db = get_db()
        cursor = db.cursor(dictionary=True)

        query = """
            SELECT 
                c.course_id, c.course_code, c.course_title, c.course_status,
                l.user_id, l.role,
                p.first_name, p.last_name,
                c.course_description, c.course_category, c.target_audience,
                c.prerequisites, c.learning_outcomes, c.duration_hours,
                c.course_fee, c.max_students, c.published,
                c.created_by, c.approved_by, c.date_created,
                c.date_updated, c.date_published, c.date_modified
            FROM courses c
            JOIN login l ON c.created_by = l.user_id
            JOIN personal_information p ON l.user_id = p.user_id
            WHERE c.course_id = %s
        """
        cursor.execute(query, (course_id,))
        course = cursor.fetchone()

        if course:
            return jsonify({
                'status': 'success',
                'course': {
                    'course_id': course['course_id'],
                    'course_code': course['course_code'],
                    'course_title': course['course_title'],
                    'course_description': course['course_description'],
                    'course_category': course['course_category'],
                    'target_audience': course['target_audience'],
                    'prerequisites': course['prerequisites'],
                    'learning_outcomes': course['learning_outcomes'],
                    'duration_hours': course['duration_hours'],
                    'course_fee': course['course_fee'],
                    'max_students': course['max_students'],
                    'course_status': course['course_status'],
                    'user_id': course['user_id'],
                    'role': course['role'],
                    'first_name': course['first_name'],
                    'last_name': course['last_name']
                }
            })
        else:
            return jsonify({'status': 'error', 'message': 'Course not found'}), 404
    except Exception as e:
        return jsonify({'status': 'error', 'message': str(e)}), 500

@admin_courses_approval_bp.route('/course/approve/<int:course_id>', methods=['POST'])
def approve_course(course_id):
    try:
        db = get_db()
        cursor = db.cursor()

        # Assuming admin user_id is stored in session as 'user_id'
        admin_user_id = session.get('user_id')

        if not admin_user_id:
            return jsonify({'status': 'error', 'message': 'Admin not logged in.'}), 401

        query = """
            UPDATE courses 
            SET course_status = 'active',
                approved_by = %s,
                date_published = %s,
                date_modified = %s
            WHERE course_id = %s
        """
        now = datetime.now()
        cursor.execute(query, (admin_user_id, now, now, course_id))
        db.commit()

        return jsonify({'status': 'success', 'message': 'Course approved successfully.'})
    except Exception as e:
        db.rollback()
        return jsonify({'status': 'error', 'message': str(e)}), 500


@admin_courses_approval_bp.route('/course/reject/<int:course_id>', methods=['POST'])
def reject_course(course_id):
    try:
        db = get_db()
        cursor = db.cursor()

        admin_user_id = session.get('user_id')

        if not admin_user_id:
            return jsonify({'status': 'error', 'message': 'Admin not logged in.'}), 401

        query = """
            UPDATE courses 
            SET course_status = 'inactive',
                approved_by = %s,
                date_modified = %s
            WHERE course_id = %s
        """
        now = datetime.now()
        cursor.execute(query, (admin_user_id, now, course_id))
        db.commit()

        return jsonify({'status': 'success', 'message': 'Course rejected successfully.'})
    except Exception as e:
        db.rollback()
        return jsonify({'status': 'error', 'message': str(e)}), 500